### PR TITLE
fixed audit issue #30 . Correction in start price in case of previous auction timed out

### PR DESCRIPTION
**Pull request by patidarmanoj10**
fixed audit issue #30 . Correction in start price in case of auction timed out.  It should be 2x of price when auction timed out instead of last purchase price.

### CODE REVIEW

**Comment by patidarmanoj10** [comment]
@rokso @filcap  Please review

File: contracts/monolithic.sol [hunk 1]
```diff
@@ -1346,7 +1346,22 @@ contract Auctions is Pricer, Owned {
...
+            if (mintable == 0 || totalAuctions == 0) {
+                _startPrice = (lastPurchasePrice * 2) + 1;   
+            } else {
+                if (currAuc == 1) {
+                    _startPrice = minimumPrice * 2;
+                } else {
+                    uint tickWhenAuctionEnded = whichTick(_startTime - 1 days);
+                    uint numTick = 0;
+                    if (tickWhenAuctionEnded > lastPurchaseTick) {
+                        numTick = tickWhenAuctionEnded - lastPurchaseTick;
```
Comment (filcap) [inline contracts/monolithic.sol]: Should we use safemath for all the math operations

File: contracts/monolithic.sol [hunk 2]
```diff
@@ -1346,7 +1346,22 @@ contract Auctions is Pricer, Owned {
         if (totalAuctions > 1) {
             _startPrice = lastPurchasePrice / 100 + 1;
         } else {
-            _startPrice = (lastPurchasePrice * 2) + 1;
+            if (mintable == 0 || totalAuctions == 0) {
```
Comment (filcap) [inline contracts/monolithic.sol]: Since lot of if conditions can we add a single comment inside each if/else branch

Comment (filcap) [reply contracts/monolithic.sol]: Also lets confirm each if / else branch is represented in our test suite

File: contracts/monolithic.sol (hunk 1 above)
Comment (patidarmanoj10) [reply contracts/monolithic.sol]: this value will never overflow so we do not need safemath.

File: contracts/monolithic.sol (hunk 2 above)
Comment (patidarmanoj10) [reply contracts/monolithic.sol]: @filcap added comment and one test case. Anyway we are going to make this function internal for upcoming commit so will add more test case then.

**Comment by filcap** [comment]
LGTM, remember to squash and merge.  thx!

**Pull request merged**